File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/repository/sample/RoleRepository.java
package com.coupang.springframework.data.requery.repository.sample;

import com.coupang.springframework.data.requery.domain.sample.Role;
import com.coupang.springframework.data.requery.repository.RequeryRepository;
import io.requery.query.Result;
import io.requery.query.Return;

import java.util.List;
import java.util.Optional;

/**
 * RoleRepository
 *
 * @author debop@coupang.com
 * @since 18. 6. 12
 */
public interface RoleRepository extends RequeryRepository<Role, Integer> {

    List<Role> findAll();

    Optional<Role> findById(Integer id);


    @Override
    Optional<Role> findOne(Return<? extends Result<Role>> whereClause);


    Long countByName(String name);
}

File: spring-data-requery-kotlin/src/test/kotlin/org/springframework/data/requery/kotlin/repository/sample/RoleRepository.kt
package org.springframework.data.requery.kotlin.repository.sample

import io.requery.query.Result
import io.requery.query.Return
import org.springframework.data.requery.kotlin.domain.sample.RoleEntity
import org.springframework.data.requery.kotlin.repository.RequeryRepository
import java.util.*

/**
 * RoleRepository
 *
 * @author debop@coupang.com
 */
interface RoleRepository: RequeryRepository<RoleEntity, Int> {

    override fun findAll(): List<RoleEntity>

    override fun findById(id: Int): Optional<RoleEntity>

    override fun findOne(filter: Return<out Result<RoleEntity>>): Optional<RoleEntity>

    fun countByName(name: String): Long
}